File: huiwen.py
def main():
	ans = 1
	i = 0
	j = 0
	k = 0
	str = [0] * (100)
	str = input()
	i = len(str)
	j = i+1
	j = j/2
	k = 0
	while k<j:
		if str[k]!=str[i-1-k]:
			ans = 0
			break
		k += 1
	print(ans)
	return 0

if __name__ == '__main__':
	main()

File: huiwen.cpp
#include <iostream>
using namespace std;

int main()
{
	int ans = 1;
	int i = 0;
	int j = 0;
	int k = 0;
	char str[100];
	cin >> str;
    i = strlen(str);
	j = i+1;
        j = j/2;
	for(k = 0;k < j;k++)
	{
		if(str[k] != str[i-1-k])
		{
			ans = false;
			break;
		}
	}
	cout << ans;
	return 0;
}